packages/compass-databases-navigation/src/databases-navigation-tree.tsx

```diff
@@ -0,0 +1,394 @@
+              'error',
+            ].includes(collectionsStatus);
+
+            const placeholdersLength = Math.min(
+              collectionsLength,
+              MAX_COLLECTION_PLACEHOLDER_ITEMS
+            );
+
+            return ([database] as TreeItem[]).concat(
+              areCollectionsReady
+                ? collections.map(({ _id: id, name, type }, index) => ({
+                    key: `${dbIndex}-${index}`,
+                    level: 2,
+                    id,
+                    name,
+                    type: type as 'collection' | 'view' | 'timeseries',
+                    setSize: collections.length,
+                    posInSet: index + 1,
+                  }))
+                : Array.from({ length: placeholdersLength }, (_, index) => ({
+                    key: `${dbIndex}-${index}`,
+                    type: 'placeholder' as const,
+                  }))
+            );
+          }
+        )
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
+        .flat()
+    );
+  }, [databases, expanded]);
+
+  const onExpandedChange = useCallback(
+    (item, isExpanded) => {
+      onDatabaseExpand(item.id, isExpanded);
+    },
+    [onDatabaseExpand]
+  );
+
+  const onFocusMove = useCallback(
+    (item) => {
+      const idx = items.indexOf(item);
+      if (idx >= 0) {
+        // It is possible that the item we are trying to move the focus to is
+        // not rendered currently. Scroll it into view so that it's rendered and
+        // can be focused
+        listRef.current?.scrollToItem(idx);
+      }
+    },
+    [items]
+  );
+
+  const [rootProps, currentTabbable] = useRovingTabIndex<HTMLDivElement>({
+    items: items as NavigationTreeData,
+    activeItemId: activeNamespace,
+    onExpandedChange,
+    onFocusMove,
+  });
+
+  const itemData: ListItemData = useMemo(() => {
+    return {
+      items,
+      isReadOnly,
+      activeNamespace,
+      currentTabbable,
+      onNamespaceAction,
+      onDatabaseExpand,
+    };
+  }, [
+    items,
+    isReadOnly,
+    activeNamespace,
+    currentTabbable,
+    onNamespaceAction,
+    onDatabaseExpand,
+  ]);
+
+  // TODO: See comment about row size in constants
+  // const getItemSize = useCallback(
+  //   (index) => {
+  //     return items[index].type === 'database'
+  //       ? DATABASE_ROW_HEIGHT
+  //       : COLLECTION_ROW_HEIGHT;
+  //   },
+  //   [items]
+  // );
+
+  const getItemKey = useCallback((index: number, data: typeof itemData) => {
+    return data.items[index].key;
+  }, []);
+
+  return (
+    <>
+      <VisuallyHidden id={id}>Databases and Collections</VisuallyHidden>
+      <div
+        role="tree"
+        aria-labelledby={id}
+        className={navigationTree}
+        {...rootProps}
+      >
+        <AutoSizer
+          disableWidth={Boolean(process.env.TEST_AUTOSIZER_DEFAULT_WIDTH)}
+          disableHeight={Boolean(process.env.TEST_AUTOSIZER_DEFAULT_HEIGHT)}
```

Comment:
In the interest of making Compass more web compatible, would it make sense to have these as optional props or have them polluting the global namespace a bit for tests instead of process env variables? Fine how it is for Compass at the moment, just a thought.

Reply:
Yeah, totally, I can move them to props

Reply:
Done in a2405d7